core/auth/identity.go

```diff
@@ -0,0 +1,26 @@
+package auth
+
+type (
+	// Identity donates an authentication object which at least identifies the authenticated subject
+	Identity interface {
+		Subject() string
```

Comment:
We are not providing a generic "User" struct anymore as a simple unified model for all other modules requiring a simple "user" ?
That would mean every other auth implementation need to add this?

(Maybe as optional extra interface like the "HasRoles" one)

Reply:
What would a generic "User" be like?

Reply:
something like User: email, firstname,lastname as a bare minimum

Reply:
We can not assume that this information will exist. HTTP auth for example only provides a username/identifier, nothing more. Also we had projects in the past where no user information but a technical ID was available. So, unless there is a standard which defines names, naming scheme, etc., we can not use it.

Also firstname and lastname will not apply for machine authentication.